src/Xamarin.Android.Build.Tasks/Tests/Xamarin.ProjectTools/Common/Builder.cs

```diff
@@ -398,8 +403,9 @@ protected bool BuildInternal (string projectOrSolution, string target, string []
 				}
 				if (RunningMSBuild) {
 					psi.EnvironmentVariables ["MSBUILD"] = "msbuild";
-					sw.WriteLine ($" /bl:\"{Path.GetFullPath (Path.Combine (XABuildPaths.TestOutputDirectory, Path.GetDirectoryName (projectOrSolution), "msbuild.binlog"))}\"");
+					sw.WriteLine ($" /bl:\"{EscapeForResponseFile (Path.Combine (projectDir, "msbuild.binlog"))}\"");
 				}
+				psi.EnvironmentVariables ["NUGET_PACKAGES"] = nugetDir;
```

Comment:
If this PR works, I should change this to use the `$(RestorePackagesPath)` MSBuild property.

Reply:
Here is a thought. Can we use envvar to set `XamarinBuildDownloadDir` ? rather than do all that nasty path stuff?